File: apiwznm/WznmQQryList.cpp
/**
	* \file WznmQQryList.cpp
	* API code for table TblWznmQQryList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 27 Aug 2020
	* \date modified: 27 Aug 2020
	*/

#include "WznmQQryList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQQryList
 ******************************************************************************/

WznmQQryList::WznmQQryList(
			const uint jnum
			, const string sref
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string stubRefWznmMVersion
			, const string stubSupRefWznmMQuery
			, const string srefSupIxVSubrole
			, const string titSupIxVSubrole
			, const string stubRefWznmMTable
			, const string stubQtbRefWznmMTable
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->stubRefWznmMVersion = stubRefWznmMVersion;
	this->stubSupRefWznmMQuery = stubSupRefWznmMQuery;
	this->srefSupIxVSubrole = srefSupIxVSubrole;
	this->titSupIxVSubrole = titSupIxVSubrole;
	this->stubRefWznmMTable = stubRefWznmMTable;
	this->stubQtbRefWznmMTable = stubQtbRefWznmMTable;
};

bool WznmQQryList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQQryList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVersion", "ver", stubRefWznmMVersion);
		extractStringUclc(docctx, basexpath, "stubSupRefWznmMQuery", "sup", stubSupRefWznmMQuery);
		extractStringUclc(docctx, basexpath, "srefSupIxVSubrole", "srl", srefSupIxVSubrole);
		extractStringUclc(docctx, basexpath, "titSupIxVSubrole", "srl2", titSupIxVSubrole);
		extractStringUclc(docctx, basexpath, "stubRefWznmMTable", "tbl", stubRefWznmMTable);
		extractStringUclc(docctx, basexpath, "stubQtbRefWznmMTable", "qtb", stubQtbRefWznmMTable);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQQryList
 ******************************************************************************/

ListWznmQQryList::ListWznmQQryList() {
};

ListWznmQQryList::ListWznmQQryList(
			const ListWznmQQryList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQQryList(*(src.nodes[i]));
};

ListWznmQQryList& ListWznmQQryList::operator=(
			const ListWznmQQryList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQQryList(*(src.nodes[i]));

	return *this;
};

ListWznmQQryList::~ListWznmQQryList() {
	clear();
};

void ListWznmQQryList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQQryList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQQryList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQQryList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQQryList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQQryList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQQryList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQQryList.java
/**
	* \file WznmQQryList.h
	* Java API code for record of table TblWznmQQryList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQQryList {
	
	public WznmQQryList(
				int jnum
				, String sref
				, String srefIxVBasetype
				, String titIxVBasetype
				, String stubRefWznmMVersion
				, String stubSupRefWznmMQuery
				, String srefSupIxVSubrole
				, String titSupIxVSubrole
				, String stubRefWznmMTable
				, String stubQtbRefWznmMTable
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.stubRefWznmMVersion = stubRefWznmMVersion;
		this.stubSupRefWznmMQuery = stubSupRefWznmMQuery;
		this.srefSupIxVSubrole = srefSupIxVSubrole;
		this.titSupIxVSubrole = titSupIxVSubrole;
		this.stubRefWznmMTable = stubRefWznmMTable;
		this.stubQtbRefWznmMTable = stubQtbRefWznmMTable;
	};

	public int jnum;
	public String sref;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String stubRefWznmMVersion;
	public String stubSupRefWznmMQuery;
	public String srefSupIxVSubrole;
	public String titSupIxVSubrole;
	public String stubRefWznmMTable;
	public String stubQtbRefWznmMTable;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQQryList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			stubRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVersion", "ver", null, 0);
			stubSupRefWznmMQuery = Xmlio.extractStringUclc(doc, basexpath, "stubSupRefWznmMQuery", "sup", null, 0);
			srefSupIxVSubrole = Xmlio.extractStringUclc(doc, basexpath, "srefSupIxVSubrole", "srl", null, 0);
			titSupIxVSubrole = Xmlio.extractStringUclc(doc, basexpath, "titSupIxVSubrole", "srl2", null, 0);
			stubRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMTable", "tbl", null, 0);
			stubQtbRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubQtbRefWznmMTable", "qtb", null, 0);
			
			return true;
		};

		return false;
	};
};

